src/event/value.rs

```diff
@@ -227,6 +227,34 @@ impl Value {
         self.as_bytes()
     }
 
+    pub fn as_option(&self) -> Option<&Value> {
+        match self {
+            Value::Null => None,
+            value => Some(value),
+        }
+    }
+
+    pub fn as_i64(&self) -> Option<i64> {
+        match self {
+            Value::Integer(i) => Some(*i),
+            _ => None,
+        }
+    }
+
+    pub fn as_map(&self) -> Option<&BTreeMap<String, Value>> {
+        match self {
+            Value::Map(v) => Some(v),
+            _ => None,
+        }
+    }
+
+    pub fn as_bytes2(&self) -> Option<Bytes> {
+        match self {
+            Value::Bytes(b) => Some(b.clone()), // cloning a Bytes is cheap
+            _ => None,
+        }
+    }
```

Comment:
"Unfortunately", we already have `as_bytes`, which behaves differently from what you'd expect (given the precedence set by Serde itself), as it tries to convert any `Value` type into `Bytes` by doing lossy conversions.

Happy to do a sweep-and-rename for the original `as_bytes` if others agree this should be the canonical implementation of such a method.

Reply:
Imo serde is wrong here. I think a better name might be something like `try_get_bytes`.

Reply:
It might be nice to have a function `as_utf8`. A very common occurrence in remap is to get the bytes and then convert to utf8.

Reply:
I agree that the existing `as_bytes` method isn't well named. Given all of the other `as_X` methods you are adding here, I'd prefer to rename the existing `as_bytes` to something else to make the `as_X` methods consistent. Maybe we could call it `to_bytes`? Or something else indicating that it is serializing the data.

In my small sample audit, it looks like `as_bytes` is only used with the `message` key or other fields expected to be `Value::Bytes` so it's possible that the other enum implementations aren't even useful there and maybe we could drop it and just use your new `as_bytes` method. I'm not 100% sure though.